run-build-functions.sh

```diff
@@ -72,6 +72,29 @@ mkdir -p $NETLIFY_CACHE_DIR/.cargo
 : ${NPM_FLAGS=""}
 : ${BUNDLER_FLAGS=""}
 
+# Run `nvm install ...`
+nvm_install() {
+  if nvm install "$@"; then
+    return 0
+  fi
+
+  # nvm install might fail due to either:
+  #  - A user error, e.g. the NODE_VERSION being an invalid version
+  #  - A third-party outage of nodejs.org/dist
+  # We distinguish between those by making a HEAD HTTP request to nodejs.org/dist
+  if curl --head --fail "$DEFAULT_NODE_MIRROR"; then
```

Comment:
`nvm` uses `curl` internally, so we re-use it too to be closer to the `nvm` logic.